### ISSUE TITLE
Correction factors

### ISSUE THREAD

**Issue by smXplorer** [Photon-HDF5/photon-hdf5#38]
It would seem useful to offer the ability to save calculated correction factors for 2-color smFRET:
- donor leakage in acceptor channel
- direct acceptor excitation by the donor laser
- gamma and beta factor

**Comment by tritemio** [comment]
Maybe.

To think about:
- Single-spot vs multispot. Per-channel or global?
- It is a piece of information whose provenance cannot be explained.
- the definition of direct-excitation may change in ALEX non-ALEX
- donor leakage and A-direct excitation is a dyes/setup property, but gamma can change from sample to sample with same dyes.
- If we include this post-analysis data why not add more?

In general I see the Photon-HDF5 file as "immutable raw data". Any post-analysis data should go in a different file (every time we write the file there is a risk of corruption). But for these 3 parameters it may be ok to do an exception.

**Comment by tritemio** [comment]
@smXplorer, after more thoughts I think it would be useful to provide these 3 fields as a standard place where to store these parameters. Even with the limitation that the source of these parameters would be not specified, a standard place is useful. They would be optional. 

These parameters depend on both sample and setup, but it is hard to choose between `sample` or `setup` group. I would to put them in `/photon_data/measurement_specs/`. For example:

- `/photon_data/measurement_specs/donor_leakage`
- `/photon_data/measurement_specs/acceptor_direct_excitation`
- `/photon_data/measurement_specs/gamma`

These fields are spot-specific and would also be compatible automatically compatible with the multispot layout. This fields  would not work for 3 or more color ALEX, and I think it's ok to leave this case out for now.

**Comment by smXplorer** [comment]
You could add a field (URL, DOI) for each parameter to specify its origin (Notebook, publication, etc).

**Comment by tritemio** [comment]
Yes, we could add a new filed ``corrections_factors_provenance`` containing an URL. It can be a DOI-URL (http://doi.org/XXXX) or any other URL (e.g. a github repository).

**Comment by tritemio** [comment]
There is the issue of which direct excitation definition to use.

Using reference: https://doi.org/10.1101/083287, the direct excitation *Dir* term can be expressed as a function of n_aa as in ALEX:

```
Dir = d_aa * n_aa
```
or as a function of physical parameters

```
Dir = d_T (gamma nd + na)
```

The link between the two is:

```
d_T = beta d_aa
```

The advantage of the second definition is that the same coefficient is valid for all measurements using the same A fluorophore and D laser (regardless of A-laser). In particular it can be used for single-laser smFRET measurements.

We can use the definition `d_T` and also add another field for `beta` so that the common ALEX definition `d_aa` can be derived.
